CDCgov/prime-reportstream
# experience/7310: make message tracker search form... a form

This PR makes the message tracker form much more accessible.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. fire up the front-end
2. (optional) fire up the back-end
3. log in to the front-end as a PrimeAdmin role user
4. enable the "message-tracker" feature flag in http://localhost:3000/admin/features
5. browse to http://localhost:3000/admin/message-tracker
6. here we can now click on the "Message Id" label and it should make the input active
7. we can also tab, and shift+tab, around the form and it should go in the proper order
8. if something is typed in to the input, pressing the "enter" should submit the form
9. the Message Id field is now required. not having a value in there should prompt the user to put in a value via HTML5 validation
10. tabbing around and pressing the "space bar key" on the "Search" and "Clear" buttons should trigger the functionality

## Changes
- make form

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7310 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #  

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
